pages/api/applications/[applicationId]/tasks/index.ts

```diff
@@ -1,5 +1,128 @@
+  [MessageType.INVALID_DUE_DATE, [{ type: StatusMessageType.Error, message: 'Task due date is invalid.' }]],
+  [
+    MessageType.INVALID_NOTIFICATION_DATETIME,
+    [{ type: StatusMessageType.Error, message: 'Task notification date and time is invalid.' }],
+  ],
+  [
+    MessageType.TASK_APPLICATION_NOT_FOUND,
+    [{ type: StatusMessageType.Error, message: 'Application for this task cannot be found.' }],
+  ],
+  [
+    MessageType.TASK_APPLICATION_DOES_NOT_BELONG_TO_USER,
+    [{ type: StatusMessageType.Error, message: 'Application for this task does not belong to you.' }],
+  ],
+  [MessageType.TASK_CREATE_SUCCESSFULLY, [{ type: StatusMessageType.Success, message: 'Task created succesfully.' }]],
+]);
+
+function handler(userId: string, req: NextApiRequest, res: NextApiResponse) {
+  const method = req.method;
+
+  switch (method) {
+    case HTTP_POST_METHOD:
+      handlePost(userId, req, res);
+      break;
+    default:
+      rejectHttpMethod(res, method);
+  }
 }
+
+async function handlePost(
+  userId: string,
+  req: NextApiRequest,
+  res: NextApiResponse<ApiResponse<TaskData | EmptyPayload>>,
+) {
+  if (!isValidRequest(req, res)) {
+    return;
+  }
+
+  const taskPostData: TaskPostData = {
+    ...req.body,
+    dueDate: new Date(req.body.dueDate),
+    notificationDateTime: req.body.notificationDateTime ? new Date(req.body.notificationDateTime) : null,
+  };
+
+  const applicationId = Number(req.query.applicationId);
+  const application = await prisma.application.findFirst({ where: { id: applicationId } });
+
+  if (!application) {
+    res
+      .status(HTTP_STATUS_NOT_FOUND)
+      .json(createJsonResponse({}, messages.get(MessageType.TASK_APPLICATION_NOT_FOUND)));
+    return;
+  }
+
+  if (application.userId !== userId) {
+    res
+      .status(HTTP_STATUS_FORBIDDEN)
+      .json(createJsonResponse({}, messages.get(MessageType.TASK_APPLICATION_DOES_NOT_BELONG_TO_USER)));
+    return;
+  }
+
+  createIfPossible(res, async () => {
+    const newTask = await prisma.task.create({
+      data: {
+        applicationId,
+        ...taskPostData,
+      },
+      select: { id: true, title: true, dueDate: true, notificationDateTime: true, isDone: true },
+    });
+
+    res
+      .status(HTTP_STATUS_CREATED)
+      .json(createJsonResponse(newTask, messages.get(MessageType.TASK_CREATE_SUCCESSFULLY)));
+  });
+}
+
+function isValidRequest(req: NextApiRequest, res: NextApiResponse<ApiResponse<EmptyPayload>>): boolean {
+  if (isEmpty(req.body.title)) {
+    res.status(HTTP_STATUS_BAD_REQUEST).json(createJsonResponse({}, messages.get(MessageType.EMPTY_TITLE)));
+    return false;
+  }
+
+  if (!isValidDate(req.body.dueDate)) {
+    res.status(HTTP_STATUS_BAD_REQUEST).json(createJsonResponse({}, messages.get(MessageType.INVALID_DUE_DATE)));
+    return false;
+  }
+
+  if (req.body.notificationDateTime && !isValidDate(req.body.notificationDateTime)) {
```

Comment:
I can pass in empty string and the request will work. It makes the field null.
```suggestion
  if (req.body.notificationDateTime !== undefined && !isValidDate(req.body.notificationDateTime)) {
```

Reply:
Would recommend checking for undefined OR null, not just undefined. E.g. if the client sends `{ notificationDateTime: null }` to mark as empty. (IIRC null is a valid http datatype)